session/session_test.go

```diff
@@ -3231,6 +3231,29 @@ func (s *testSessionSuite2) TestPerStmtTaskID(c *C) {
 	c.Assert(taskID1 != taskID2, IsTrue)
 }
 
+func (s *testSessionSuite) TestDoDDLJobQuit(c *C) {
+	// test https://github.com/pingcap/tidb/issues/18714, imitate DM's use environment
+	store, err := mockstore.NewMockStore(mockstore.WithStoreType(mockstore.MockTiKV))
+	c.Assert(err, IsNil)
+	session.SetSchemaLease(10 * time.Millisecond)
+	dom, err := session.BootstrapSession(store)
+	c.Assert(err, IsNil)
+	se, err := session.CreateSession(store)
+	c.Assert(err, IsNil)
+
+	c.Assert(failpoint.Enable("github.com/pingcap/tidb/ddl/storeCloseInLoop", `return(2)`), IsNil)
+	go func() {
+		time.Sleep(time.Second)
```

Comment:
I think use `Sleep` and `go` will make the test unstable and "sleep 2s" make the test run a long test.